app/Models/Signup.php

```diff
@@ -64,6 +64,22 @@ public function posts()
         return $this->hasMany(Post::class)->with('tags');
     }
 
+    /**
+     * Get the visible posts associated with this signup.
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function visiblePosts()
+    {
+        if (! is_staff_user()) {
+            return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+                                              ->orWhere('northstar_id', auth()->id())
+                                              ->with('tags');
+        }
+
+        return $this->hasMany(Post::class)->with('tags');
+    }
+
```

Comment:
@sbsmith86 continuing from [this comment](https://github.com/DoSomething/rogue/pull/689#discussion_r184439169), calling `Signup::with('visiblePosts')->find($signup->id);` returns all the posts for a non staff/admin/owner user. 

It seems like it goes through the `posts()` method in `Signup.php` because when I place a `dd()` there, it reaches it and I'm not sure how to avoid this?

Reply:
@chloealee have you pushed up the `visiblePosts` method? would like to see what is happening so I can follow along.

Reply:
@sbsmith86 yup! it is this method! :)

Reply:
@sbsmith86 no worries it is hard to see when it cuts off like that! 

Yes! I tried doing `$signup = $signup->visiblePosts()` and it gave me this error: 

```
Type error: Argument 1 passed to Rogue\Http\Transformers\SignupTransformer::transform() must be an instance of Rogue\Models\Signup, instance of Illuminate\Database\Eloquent\Relations\HasMany given, called in /home/vagrant/DoSomething.org/rogue/vendor/league/fractal/src/Scope.php on line 338
``` 
since it seems like it is passing the relationship, not the Signup model, to the transformer [here](https://github.com/DoSomething/rogue/pull/689/files#diff-28d9d92df597ce839f7f0cfc3a8dfb94R127). 

Same thing happened when I deleted the `with('tags');` :-/ 

I'm not sure what else I might be doing wrong?

Reply:
where were you calling `$signup = $signup->visiblePosts()`? I think you should be doing this in the transformer itself when deciding to embed posts or not, so i'm tryna figure out why the transformer is being mad it isn't given the right thing.

Reply:
OMG @sbsmith86 that was it! I was calling that in the controller 🤦‍♀️ !! This makes TOTAL sense now thank you so much!!! I'll push up the changes now!

Reply:
YAY so glad we figured it out. And I am sorry I made you dig all the way through that hahaha

Reply:
no this was really good to get me to better understand the relation stuff that I thought I had a handle on! Thank you so much for your patience with me and going through this so much!! :)

Reply:
pr is updated with adding to the transformer!